### ISSUE TITLE
Textfield in loop not receiving styling

### ISSUE THREAD

**Issue by nathanjohnson320** [debois/elm-mdl#250]
I edited the many-counter example and added

```
, Textfield.render Mdl
            [ 2, idx ]
            model.mdl
            [ Textfield.label "?????" ]
        ]
```

to line 100 but when I start this up in elm reactor the textfield doesn't have styling and the label doesn't float above when you focus it.

**Comment by OvermindDL1** [comment]
> to line 100 but when I start this up in elm reactor the textfield doesn't have styling and the label doesn't float above when you focus it.

Do the other textfields still have styling and floating?  Only reason that should not work is if the CSS did not apply (there is no javascript or so associated with that, it is pure css).

**Comment by nathanjohnson320** [comment]
Interesting. Below is the full example, if I add a textfield inside of the view function it works fine but where I currently have it the styles are not attached.

```
{- This file re-implements the Elm Counter example (many counters) with
   elm-mdl. Look at this file if you have a dynamic number of elm-mdl components
   and are unsure how to choose proper indices.
   If you are looking for a starting point, you want `Counter.elm` rather than
   this file.
-}
module Main exposing (..)
import Html.App as App
import Html exposing (..)
import Html.Attributes exposing (href, class, style)
import Array exposing (Array)
import Material
import Material.Scheme
import Material.Button as Button
import Material.Textfield as Textfield
import Material.Options exposing (css)
import Material.Helpers exposing (pure)
-- MODEL
type alias Model =
    { counters : Array Int
    , mdl : Material.Model
    }
model : Model
model =
    { counters = Array.empty
    , mdl = Material.model
    }
-- ACTION, UPDATE
type Msg
    = Increase Int
    | Reset Int
    | Add
    | Remove
    | Mdl (Material.Msg Msg)
map : Int -> (a -> a) -> Array a -> Array a
map k f a =
    Array.get k a
        |> Maybe.map (\x -> Array.set k (f x) a)
        |> Maybe.withDefault a
update : Msg -> Model -> ( Model, Cmd Msg )
update msg model =
    case msg of
        Increase k ->
            pure { model | counters = map k ((+) 1) model.counters }
        Reset k ->
            pure { model | counters = map k (always 0) model.counters }
        Add ->
            pure { model | counters = Array.push 0 model.counters }
        Remove ->
            pure { model | counters = Array.slice 0 (Array.length model.counters - 1) model.counters }
        Mdl msg' ->
            Material.update msg' model
-- VIEW
type alias Mdl =
    Material.Model
view1 : Int -> Int -> Html Msg
view1 idx val =
    div [ style [ ( "padding", "2rem" ) ] ]
        [ text ("Current count: " ++ toString val)
        , Button.render Mdl
            [ 0, idx ]
            model.mdl
            [ Button.onClick (Increase idx)
            , css "margin" "0 24px"
            ]
            [ text "Increase" ]
        , Button.render Mdl
            [ 1, idx ]
            model.mdl
            [ Button.onClick (Reset idx) ]
            [ text "Reset" ]
        , Textfield.render Mdl
            [ 2, idx ]
            model.mdl
            [ Textfield.label "?????" ]
        ]
view : Model -> Html Msg
view model =
    let
        counters =
            model.counters
                |> Array.toList
                |> List.indexedMap view1
    in
        List.concatMap identity
            [ [ Button.render Mdl
                    [ 2 ]
                    model.mdl
                    [ Button.onClick Add ]
                    [ text "Add counter" ]
              ]
            , [ Button.render Mdl
                    [ 4 ]
                    model.mdl
                    [ Button.onClick Remove ]
                    [ text "Remove counter" ]
              ]
            , counters
            ]
            |> div []
            |> Material.Scheme.top
main : Program Never
main =
    App.program
        { init = ( model, Cmd.none )
        , view = view
        , subscriptions = always Sub.none
        , update = update
        }
```

**Comment by OvermindDL1** [comment]
Hmm, does it happen on a generated site or only in the elm-reactor?  This seems like one of those many elm-reactor bugs/limitations...

**Comment by nathanjohnson320** [comment]
I thought it was specific to the reactor but even when I output this to js and then include it in the below HTML it still doesn't render the textfield properly.

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    <title>The elm-mdl library</title>

    <link href='https://fonts.googleapis.com/css?family=Roboto:400,300,500|Roboto+Mono|Roboto+Condensed:400,700&subset=latin,latin-ext' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    <link rel="stylesheet" href="https://code.getmdl.io/1.2.0/material.teal-red.min.css" />
    <script src="https://cdn.polyfill.io/v2/polyfill.js?features=Event.focusin"></script>

  </head>
  <body>
    <!-- elm -->
    <script src="./elm.js"></script>
    <script>
      app = Elm.Main.fullscreen();
    </script>
  </body>
</html>
```

![image](https://cloud.githubusercontent.com/assets/1680341/20021664/3b05f9c4-a2ad-11e6-84ba-6abb24431066.png)

**Comment by OvermindDL1** [comment]
Now that is indeed funky, can you trace the css and where it is coming from at that point (or host it somewhere so we can try)?

**Comment by nathanjohnson320** [comment]
Here's the hosted version: https://obscure-brushlands-18096.herokuapp.com/
And the code behind it: https://github.com/nathanjohnson320/elmmdltextfield

**Comment by OvermindDL1** [comment]
Ah, it is not getting set the floating class.

And I'm not seeing in the code where you are setting `Textfield.floatingLabel`?

**Comment by nathanjohnson320** [comment]
Sorry I just added the floating label back I was testing a bunch of different combinations.

https://github.com/nathanjohnson320/elmmdltextfield/blob/master/elm/Main.elm#L101

**Comment by nathanjohnson320** [comment]
For some reason it's not receiving the "is-focused" class and that's causing the unstyled problems.

**Comment by OvermindDL1** [comment]
Yeah and I'm not seeing a javascript event handler attached to it either, something is bugged.

Oh devs where be ye?  ^.^

**Comment by debois** [comment]
@nathanjohnson320: Could you re-check whether this is still a problem in 8.0.0? (Focus-management for textfield changed relatively to 7.x.x.)

**Comment by nathanjohnson320** [comment]
Yes this appears to be resolved in elm 0.18 + elm-mdl 8, thanks!
